packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -54,6 +55,9 @@ export interface DialogProps
   /** A dialog can contain a header. */
   header?: ShorthandValue
 
+  /** A dialog can contain a button next to the header. */
+  headerAction?: ShorthandValue
```

Comment:
```suggestion
  headerAction?: ShorthandValue<ButtonProps>
```

🙏

Reply:
Added for all ShorthandValue props in the Dialog. :)